src/compiler/scala/tools/nsc/typechecker/SyntheticMethods.scala

```diff
@@ -307,13 +314,13 @@ trait SyntheticMethods extends ast.TreeDSL {
       Any_equals -> (() => equalsDerivedValueClassMethod)
     )
 
-    def caseClassMethods = productMethods ++ /*productNMethods ++*/ Seq(
+    def caseClassMethods = productClassMethods ++ /*productNMethods ++*/ Seq(
```

Comment:
we can remove the intermediate collections easily here either in this PR or another - just to remove some CPU/memory in the compile